src/capture/image/png_writer.cpp

```diff
@@ -242,7 +242,7 @@ void PngWriter::WritePngInfo(const uint16_t width, const uint16_t height,
 void PngWriter::WriteRow(std::vector<uint8_t>::const_iterator row)
 {
 	assert(png_ptr);
-	png_write_row(png_ptr, &*row);
+	png_write_row(png_ptr, const_cast<png_bytep>(&*row));
```

Comment:
Thanks for this fix up, @sulix.

It would be nice to operate on PNG's desired type directly, given this is a PNG-specific class ("PngWrite"), which also makes good use of other PNG-data types in its members.

In other words - because it's PNG-specific, we don't have try to be 'generic' so to speak on its interfaces (well, at least just at this point).

Reply:
(I've addressed my comment in a PR https://github.com/dosbox-staging/dosbox-staging/pull/2741)

Reply:
Going with this original solution then, as discussed.